### ISSUE TITLE
Validation on the test set

### ISSUE THREAD

**Issue by mshnmshn** [akamaster/pytorch_resnet_cifar10#35]
In the code the validation is performed on the test. This methodology is pathological and does not meet the paper statement on (45000, 5000) train/val split.
https://github.com/akamaster/pytorch_resnet_cifar10/blob/d5489e8995e81e91ce6b1d69dcc98ad579b0b153/trainer.py#L101
